File: recordOfProblems/BinaryTreeLevelOrderTraversal.java
public List<List<Integer>> levelOrder(TreeNode root) {
    Queue<TreeNode> queue = new LinkedList<>();
    List<List<Integer>> solution = new LinkedList<>();
    
    if(root != null) {
        queue.add(root);
        while(!queue.isEmpty()) {
            int levelSize = queue.size();
            List<Integer> currentLevel = new ArrayList<>();
            for(int i=0; i<levelSize; i++) {
                TreeNode current = queue.poll();
                //Process node:
                currentLevel.add(current.val);
                //Add childs
                if(current.left != null) queue.add(current.left);
                if(current.right != null) queue.add(current.right);
            }
            solution.add(currentLevel);
        }
    }
    
    return solution;
}

File: recordOfProblems/c++/BinaryTreeLevelOrderTraversal.cc
vector<vector<int>> levelOrder(TreeNode* root) {
    std::queue<TreeNode*> queue;
    std::vector<std::vector<int>> sol;
    if(root == nullptr) return sol;
    
    queue.push(root);
    while(!queue.empty()) {
        int currSize = queue.size();
        std::vector<int> currLevel;
        for(int i=0; i<currSize; i++) {
            TreeNode* currNode = queue.front();
            queue.pop();
            currLevel.push_back(currNode->val);
            if(currNode->left != nullptr) queue.push(currNode->left);
            if(currNode->right != nullptr) queue.push(currNode->right);
        }
        sol.push_back(currLevel);
    }
    
    return sol;
}
